### PR TITLE
tarsnapper: add meta

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
- Add `meta`
- Use `pname` instead of `name`
- Add `pythonImportsCheck` and switch to `pytestCheckHook`

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 135242` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>tarsnapper</li>
  </ul>
</details>

File: pkgs/tools/backup/tarsnapper/default.nix [hunk 1]
```diff
@@ -1,25 +1,44 @@
...
 
-  propagatedBuildInputs = with python3Packages; [ pyyaml python-dateutil pexpect ];
-
-  patches = [ ./remove-argparse.patch ];
+  patches = [
+    # Remove standard module argparse from requirements
+    ./remove-argparse.patch
+  ];
 
   makeWrapperArgs = ["--prefix PATH : ${tarsnap}/bin"];
```
Comment (SuperSandro2000) [inline pkgs/tools/backup/tarsnapper/default.nix:34]:
```suggestion
  makeWrapperArgs = [ "--prefix PATH : ${lib.makeBinPath [ tarsnap ]}" ];
```

File: pkgs/tools/backup/tarsnapper/default.nix [hunk 2]
```diff
@@ -1,25 +1,44 @@
...
+    pyyaml
+    python-dateutil
+    pexpect
+  ];
 
-  checkPhase = ''
-    py.test .
-  '';
+  checkInputs = with python3Packages; [
+    nose
```
Comment (SuperSandro2000) [inline pkgs/tools/backup/tarsnapper/default.nix:25]: Are the tests executed with nose upstream?

File: pkgs/tools/backup/tarsnapper/default.nix (hunk 1 above)
Comment (fabaff) [reply pkgs/tools/backup/tarsnapper/default.nix]: Thanks, updated.

File: pkgs/tools/backup/tarsnapper/default.nix (hunk 2 above)
Comment (fabaff) [reply pkgs/tools/backup/tarsnapper/default.nix:25]: They are. I removed `pytestCheckHook`.

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 135242` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>tarsnapper</li>
  </ul>
</details>

**Pull request merged**